torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py

```diff
@@ -168,83 +166,45 @@ def _annotate_linear(
-            # this.
-            weight_node_users = list(
-                filter((lambda x: (_is_sym_size_node(x) is False)), weight_node.users)
+    for node in gm.graph.nodes:
+        if node.op != "call_function" or node.target != torch.ops.aten.linear.default:
+            continue
+        if filter_fn and not filter_fn(node):
+            continue
+        annotated_partitions.append([node])
+        act_node = node.args[0]
+        weight_node = node.args[1]
+        bias_node = None
+        if len(node.args) < 2:
+            raise ValueError(
+                f"aten.linear nodes less than 2 args which are: {node.args}"
+            )
+        if len(node.args) > 2:
+            bias_node = node.args[2]
+        # find use of act node within the matched pattern
+
+        if _is_annotated([node]) is False:  # type: ignore[list-item]
+            _annotate_input_qspec_map(
+                node,
+                act_node,
+                input_act_qspec,
             )
-            weight_use_node_in_p = set(weight_node_users).intersection(set(p.nodes))
-            if len(weight_use_node_in_p) != 1:
-                raise ValueError(
-                    f"Could not find a valid use of act node. All uses {weight_use_node_in_p}"
-                )
-            weight_use_node = weight_use_node_in_p.pop()
-
-            act_node_users = list(
-                filter((lambda x: (_is_sym_size_node(x) is False)), act_node.users)
+            _annotate_input_qspec_map(
+                node,
+                weight_node,
+                weight_qspec,
             )
-            act_use_node_in_p = set(act_node_users).intersection(set(p.nodes))
-            if len(act_use_node_in_p) != 1:
-                raise ValueError(
-                    f"Could not find a valid use of act node. All uses {act_use_node_in_p}"
-                )
-            act_use_node = act_use_node_in_p.pop()
-            if _is_annotated([act_use_node]) is False:  # type: ignore[list-item]
-                _annotate_input_qspec_map(
-                    act_use_node,
-                    act_node,
-                    input_act_qspec,
-                )
-            if bias_node and _is_annotated([bias_node]) is False:
-                _annotate_output_qspec(bias_node, bias_qspec)
-            if _is_annotated([weight_node]) is False:  # type: ignore[list-item]
-                _annotate_input_qspec_map(
-                    weight_use_node,
-                    weight_node,
-                    weight_qspec,
-                )
-            if _is_annotated([output_node]) is False:
-                _annotate_output_qspec(output_node, output_act_qspec)
-            nodes_to_mark_annotated = list(p.nodes)
-            _mark_nodes_as_annotated(nodes_to_mark_annotated)
+            _annotate_output_qspec(node, output_act_qspec)
+
+        if bias_node and _is_annotated([bias_node]) is False:
```

Comment:
why are we annotating output of bias? shouldn't it be input to aten.linear?

Reply:
yeah i just forgot to update this. It can also be annotated as input.